### PR TITLE
Adding in error handling - not finished

### PR DESCRIPTION
**Pull request by chillilauren**
You can see in the code that I am asking for a specific error if getRoverImages doesn't work, and therefore the variable images shouldn't exist (I assume). But when I check this it comes back with a different error than the one I have specified, one that has been defined in nasaApiClient.ts

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {getRoverImages} from "./services/nasaService";
 import {NewEditorRequest} from "./models/requestModels";
 import {createEditor} from "./services/authService";
 
+require('express-async-errors');
```
Comment (MichaelWalker) [inline src/app.ts]: we should probably `import` this instead

File: src/app.ts [hunk 2]
```diff
@@ -22,12 +23,37 @@ app.get('', async(request, response) => {
...
+    }
     response.json(images);
 })
+app.use((err:any, req:any, res:any, next:any) => {
+    if (err.message) {
+        res.status(500);
+        res.json({error: err.message});
+    }
+    next(err)
+})
```
Comment (MichaelWalker) [inline src/app.ts]: I don't think we need this if we've got the one above

File: src/app.ts [hunk 3]
```diff
@@ -22,12 +23,37 @@ app.get('', async(request, response) => {
...
+app.get("/test", async (res, rep) => {
+    throw Error("Oh Dear :(");
+});
+app.use((err:any, req:any, res:any, next:any) => {
+    if (err.message) {
+        res.status(500);
+        res.json({error: err.message});
+    }
+    next(err)
+})
```
Comment (MichaelWalker) [inline src/app.ts]: this looks pretty good to me.
I guess this should catch basically anything.

If we want to be able to narrow it down any more then we can enhance it later.

**Review by MichaelWalker — CHANGES_REQUESTED** [review]
This is looking pretty good to me!
Looks like it should pretty much work.

If you can tidy it up a little I'd be happy to merge! :)

File: src/app.ts (hunk 1 above)
Comment (chillilauren) [reply src/app.ts]: I tried this before but wasn't sure what to write, tried this but it doesn't like it:
`import expressAsyncErrors from 'express-async-errors';`

File: src/app.ts (hunk 3 above)
Comment (chillilauren) [reply src/app.ts]: I have ended up commenting out the bit 
`if (!images) {
     throw new Error('500: ApiError')
 }`

as this was doing nothing... in the network tab if the api doesn't work it sends a 404 code, but then that's not what it shows on the page for users to actually see

File: src/app.ts (hunk 1 above)
Comment (MichaelWalker) [reply src/app.ts]: you should just be able to do `import 'express-async-errors';`

File: src/app.ts [hunk 4]
```diff
@@ -22,10 +24,12 @@ app.get('', async(request, response) => {
     response.json(status);
 });
 
-
 app.get("/api/rovers/:name/images", async (request, response) => {
     const roverName = request.params.name;
     const images = await getRoverImages(roverName);
+    // if (!images) {
+    //     throw new Error('500: ApiError')
+    // }
```
Comment (MichaelWalker) [inline src/app.ts]: could you remove these commented lines?

File: src/nasa/nasaApiClient.ts [hunk 5]
```diff
@@ -22,6 +22,9 @@ interface RoverPhotosApiData {
 
 export const getRovers = async (): Promise<RoverApiData[]> => {
     const apiData = await get<RoversApiData>("/rovers");
+
+    console.log(apiData);
+
```
Comment (MichaelWalker) [inline src/nasa/nasaApiClient.ts]: probably don't need this either.

**Review by MichaelWalker — APPROVED** [review]

**Comment by chillilauren** [comment]
This says the checks are failing on ApiRoutes.ts which is not something I have added in here myself. I think this was a merge conflict.. I'm not sure how to fix this.

File: src/app.ts [hunk 6]
```diff
@@ -36,6 +39,12 @@ app.get('', async(request, response) => {
     response.json(status);
 });
 
+app.get("/api/rovers/:name/images", async (request, response) => {
+    const roverName = request.params.name;
+    const images = await getRoverImages(roverName);
+    response.json(images);
+})
+
```
Comment (MichaelWalker) [inline src/app.ts]: I don't think you meant to add this...
(its been moved to another file, so we don't want to add it back to here!)

**Review by MichaelWalker — CHANGES_REQUESTED** [review]

**Review by MichaelWalker — APPROVED** [review]

**Pull request merged**